src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,280 @@
+
+            String start = startAndEnd[0].trim();
+            String end = startAndEnd[1].trim();
+            LocalDateTime formattedStartDateTime = LocalDateTime.parse(start, dateTimeFormat);
+            LocalDateTime formattedEndDateTime = LocalDateTime.parse(end, dateTimeFormat);
+            return new Event(details, formattedStartDateTime, formattedEndDateTime);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid event time format. Please use dd-MM-yyyy HHmm format!");
+        }
+    }
+
+    /**
+     * Deletes a task from the list based on the given task number.
+     *
+     * @param args The task number to be deleted.
+     * @return The deleted Task object.
+     * @throws DukeException If the task number is invalid or in an incorrect format.
+     */
+    public Task deleteTask(String args) throws DukeException {
+        int taskNumber;
+        Task deletedTask;
+
+        try {
+            taskNumber = Integer.parseInt(args);
+
+            if (taskNumber <= 0) {
+                throw new DukeException("Number must be more 1 or more!");
+            } else if (this.exceedsSizeOfTaskList(taskNumber)) {
+                throw new DukeException("Number is higher than current size of task list!");
+            }
+
+            deletedTask = this.tasks.remove(taskNumber - 1);
+            this.numTasks--;
+
+            return deletedTask;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Deleting task should be in this format: delete [task number]");
+        }
+    }
+
+    /**
+     * Changes the done status of a task based on the given task number and state.
+     *
+     * @param args The task number to be marked/unmarked and the state.
+     * @param toBeDone The desired state (true for done, false for undone).
+     * @return The Task object with the modified done status.
+     * @throws DukeException If the task number is invalid or in an incorrect format.
+     */
+    public Task changeTaskDoneStatus(String args, boolean toBeDone) throws DukeException {
+        int taskNumber;
+
+        try {
+            taskNumber = Integer.parseInt(args);
+
+            if (taskNumber <= 0) {
+                throw new DukeException("Number must be more 1 or more!");
+            } else if (this.exceedsSizeOfTaskList(taskNumber)) {
+                throw new DukeException("Number is higher than current size of task list!");
+            }
+
+            Task taskToChange = this.tasks.get(taskNumber - 1);
+
+            if (toBeDone) {
+                taskToChange.markDone();
+            } else {
+                taskToChange.markUndone();
+            }
+            return taskToChange;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Marking/unmarking tasks should be in this format: mark/unmark [task number]");
+        } catch (DukeException e) {
+            throw new DukeException(e.getMessage());
+        }
+    }

```

Comment:
As with several other instances, perhaps handling all the steps of parsing, validation, and error handling through calls to other methods will make these methods more adherent to the SLAP principle.